mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java

```diff
@@ -333,18 +334,36 @@ public void multimailboxSearchShouldReturnUidOfMessageMarkedAsSeenInTwoMailboxes
                     .from(searchQuery)
                     .inMailboxes(mailbox.getMailboxId(), mailbox2.getMailboxId())
                     .build();
-        Map<MailboxId, Collection<MessageUid>> actual = messageSearchIndex.search(session, query);
-        assertThat(actual.entrySet()).hasSize(2);
-        assertThat(actual.get(mailbox.getMailboxId())).containsExactly(m6);
-        assertThat(actual.get(mailbox2.getMailboxId())).containsExactly(m1);
+
+        List<MessageId> actual = messageSearchIndex.search(session, query, LIMIT);
+
+        assertThat(actual).containsOnly(mOther.getMessageId(), m6.getMessageId());
     }
-    
+
+    @Test
+    public void multimailboxSearchShouldLimitTheSize() throws MailboxException {
+        SearchQuery searchQuery = new SearchQuery();
+        searchQuery.andCriteria(SearchQuery.flagIsSet(Flags.Flag.SEEN));
+        MultimailboxesSearchQuery query =
+            MultimailboxesSearchQuery
+                .from(searchQuery)
+                .inMailboxes(mailbox.getMailboxId(), mailbox2.getMailboxId())
+                .build();
+
+        long limit = 1;
+        List<MessageId> actual = messageSearchIndex.search(session, query, limit);
+        // Two messages matches this query : mOther and m6
```

Comment:
Ideally your message names should carry the information.
But ok, you can keep this one (I read too fast and was confused by the 2 messages / size(1))